samples/dlopen/Dlopen.java

```diff
@@ -30,46 +30,44 @@
  */
 
 import java.lang.invoke.*;
+import java.lang.foreign.*;
 import java.util.Optional;
-import jdk.incubator.foreign.*;
+import java.util.function.Function;
 import org.unix.dlfcn_h.*;
 import static org.unix.dlfcn_h.*;
 
 public class Dlopen {
-    // implementation of SymbolLookup that loads a given shared object using dlopen
+    // implementation of Symbol lookup that loads a given shared object using dlopen
     // and looks up symbols using dlsym
-    private static SymbolLookup lookup(String libraryName, ResourceScope scope) {
-        try (ResourceScope openScope = ResourceScope.newConfinedScope()) {
-            var openScopeAllocator = SegmentAllocator.nativeAllocator(openScope);
-            var handle = dlopen(openScopeAllocator.allocateUtf8String(libraryName), RTLD_LOCAL());
+    private static Function<String, Optional<MemorySegment>> lookup(String libraryName, MemorySession session) {
```

Comment:
Does this sample still make sense with the new `libraryLookup` ?

Reply:
May be not. But as a POSIX specific example of dlopen/dlsym/dladdr etc. We may be expand to use flags, dladdr etc.